pkg/querier/metadata_handler.go

```diff
@@ -48,24 +48,28 @@ type metadataErrorResult struct {
 // Mimir for a given tenant. It is kept and returned as a set.
 func NewMetadataHandler(m MetadataSupplier) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		limit := -1
+		limit := int32(-1)
```

Comment:
Note to reviewers: changed the "base" data type to be `int32` so that casting now is always to bigger ones (except for the value returned by `strconv.ParseInt()` which is a explicitly parsed as 32bit).